### DISCUSSION TITLE
Mosaicing of many aerial photographs

### DISCUSSION THREAD

**Discussion by sidtupper** [OSGeo/grass#2608] [Q&A] [answered]
I have about 50 photos of ground imagery acquired with a drone on a regular pattern of waypoints.  Would GRASS be a good tool to mosaic the images?

**Comment by ninsbl** [comment]
Depends on your data acquisition  and aim. You could patch (mosaic) images using GRASS GIS with [r.patch](https://grass.osgeo.org/grass82/manuals/r.patch.html) or [r.buildvrt](https://grass.osgeo.org/grass82/manuals/r.buildvrt.html) after importing with r.in.gdal or r.external. If you have sufficient overlap between images, I would rather use [OpenDroneMap](https://www.opendronemap.org/) as that is specific for the purpose and also open source...

**Reply by sidtupper** [reply]
Thanks, I will look into OpenDroneMap.

**Comment by BadAssassin** [comment, marked as answer]
GRASS is capable of mosiaicing images into an orthophoto, but I would 
suggest looking into WebODM[1]. It uses newer structure-from-motion 
algorithms and I have personally had very good success using it. You 
will need to do a little tweaking to get straight edges on buildings, 
but the forums are very helpful.

They have a docker image that takes most of the installation headaches 
away and is designed with drones in mind. I often do my initial 
processing in WebODM, then move the products to GRASS for further 
development. Point clouds and orthos generated prove particularly useful 
and can seamlessly tie into DroneDB.

[1] https://www.opendronemap.org/webodm/

On 10/23/2022 11:46 AM, sidtupper wrote:
>
> I have about 50 photos of ground imagery acquired with a drone on a 
> regular pattern of waypoints. Would GRASS be a good tool to mosaic the 
> images?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/OSGeo/grass/discussions/2608>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AQZTHVNZ64SBXFPZYT365GLWEWBZJANCNFSM6AAAAAARMMZI5E>.
> You are receiving this because you are subscribed to this 
> thread.Message ID: ***@***.***>
>
-- 
Best Regards,
-Brad

**Reply by sidtupper** [reply]
Thanks!
